sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/GenericResourceImpl.java

```diff
@@ -119,32 +121,54 @@ public String managedBy() {
                 this.apiVersion());
     }
 
+    @Override
+    public GenericResourceImpl update() {
+        this.updateParameter = new GenericResourceInner();
+        return super.update();
+    }
+
     public GenericResourceImpl withProperties(Object properties) {
-        innerModel().withProperties(properties);
+        if (isInCreateMode()) {
+            innerModel().withProperties(properties);
+        } else {
+            updateParameter.withProperties(properties);
+        }
```

Comment:
This piece of code appears multiple times. They follow same structure. Is it possible to simplify it?

Reply:
Good point!

In this resource, actually we can (previously not making it mainly to avoid affect code for create flow).

Here type of create parameter = type of update parameter != type of response resource. So we can use same object for create and update parameter.
Done the change in commit https://github.com/Azure/azure-sdk-for-java/pull/23791/commits/a2656c18e74bda30b5ee225254ef0698314112a1.

Though for more common case that create parameter != update parameter, it will be harder to simplify the code.